### PR TITLE
Fix previewing changes in circle artefacts

### LINKED ISSUE #268: Preview changes using circle artefacts
Circle publishes the build artefacts on a static web server, so it should be simple to use that for QA and previewing changes.

### PR DESCRIPTION
**Pull request by bjgordon**
After circleci builds each branch, it publishes the artefacts in a static web server. With these tweaks, assets will work when viewing these artefacts. This means that links to the pages on circle can be shared to help people preview the content within the context of the site. 

e.g. https://1166-71211972-gh.circle-artifacts.com/0/home/ubuntu/dta-website/_site/index.html

image

Unfortunately, clicking on links to other parts of the site do not work, as [the circle web server does not automatically append /index.html](https://discuss.circleci.com/t/circle-artifacts-com-not-using-index-html/320) like the site expects (as it does when served by nginx) However being able to share a link to the built content on circle may still be useful.

Closes #268

### CODE REVIEW

File: bin/buildrc [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
```
Comment (bjgordon) [inline bin/buildrc]: Maybe there's a more standard filename than `buildrc`?

**Comment by klepas** [comment]
I can’t really assess all the bash goodness (maybe something for @tessereth?), but it looks good.

Given the caveat you pointed out @bjgordon, how useful do you reckon this is, eg. if someone is doing review work and isn’t aware of this limitation will they see the suggested work as being broken?

**Comment by bjgordon** [comment]
@klepas The vast majority of the time I do a `cf push` or a screenshot for the content authors before merging their PR to `develop`, there's always something to fix and the sooner they can see a preview the better. If I can teach people where to go in circleci to view the artefacts, it's one less thing they need me for. If people are regularly authoring content they'd be able to do figure it out. 
It's better than nothing, something like QA Fire would be even better of course.

**Comment by bjgordon** [comment]
Actually re your point with the broken links and people then blaming the change... Good point. If the preview link is described as just a preview for that one page, that would hopefully limit that.

File: bin/buildrc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  ${DEPLOY_BRANCHES})
+    export DTA_SITE_URL="https://dta-website-${GITBRANCH}.apps.staging.digital.gov.au"
+    ;;
+  *)
+    if [[ -n ${CIRCLECI+x} ]]
+    then
+      # Configure jekyll to be statically hosted on circleci.
+      # TODO remove hardcoded parameters as much as possible - but where does 71211972 come from?
+      export DTA_SITE_URL="https://${CIRCLE_BUILD_NUM}-71211972-gh.circle-artifacts.com"
+      export DTA_SITE_BASEURL="/${CIRCLE_NODE_INDEX}${HOME}/${CIRCLE_PROJECT_REPONAME}/_site"
```
Comment (tessereth) [inline bin/buildrc]: I think you can replace `${CIRCLE_NODE_INDEX}${HOME}/${CIRCLE_PROJECT_REPONAME}` with `$CIRCLE_ARTIFACTS`. Or at least that's what I did when I tried this: https://github.com/AusDTO/dto-design-guide/pull/26/files#diff-32d9582cfc1864479f16a9482c7ba5ecR22

**Comment by bjgordon** [comment]
Thanks for looking at it @tessereth and @klepas. Tess I tried using CIRCLE_ARTIFACTS but that got resolved to `/tmp/circle-artifacts.M8tTTjO/`, so I dont think that would work...

**Comment by tessereth** [comment]
Interesting. I was specifically outputting the build to CIRCLE_ARTIFACTS. But you are leaving it in the home directory and adding that as an artifact directory in circle.yml. I did not know you could do it that way. Cool.

**Pull request merged**